library/src/main/java/com/dslplatform/json/processor/Analysis.java

```diff
@@ -1019,6 +1028,29 @@ private void analyzeAttribute(
 			CompiledJson.TypeSignature typeSignature = typeSignatureValue(annotation);
 			JsonAttribute.IncludePolicy includeToMinimal = includeToMinimalValue(annotation);
 
+			boolean notNull = hasNonNullable(element, field, annotation);
+			boolean mandatory = hasMandatoryAnnotation(element, annotation) || field != null && hasMandatoryAnnotation(field, null);
+			if (converter != null && converter.binder != null) {
+				// if value is null, the binding value will be lost
+				if (!notNull) {
+					hasError = true;
+					messager.printMessage(
+							Diagnostic.Kind.ERROR,
+							"Attribute with binder can not be nullable on " + name + " in " + info.name,
+							element,
+							annotation);
+				}
+				// if value is omitted, we will need to have `reset` method

```

Comment:
I gave this some thought. Maybe we can get away from reset by initially assigning all values.
This way reset can be optional, but we should certainly assign all the default values before starting with bind.

Reply:
>  we should certainly assign all the default values before starting with bind.
I don't follow. Do you say that we should assign default values before each `bind` call?